src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSourceHelper.Windows.cs

```diff
@@ -14,33 +14,25 @@ internal static int GetCpuUsage()
         {
             // Returns the current process' CPU usage as a percentage
 
-            int cpuUsage;
+            int cpuUsage = 0;
 
-            if (!Interop.Kernel32.GetProcessTimes(Interop.Kernel32.GetCurrentProcess(), out _, out _, out long procKernelTime, out long procUserTime))
-            {
-                return 0;
-            }
-
-            if (!Interop.Kernel32.GetSystemTimes(out _, out long systemUserTime, out long systemKernelTime))
-            {
-                return 0;
-            }
-
-            if (s_prevSystemUserTime == 0 && s_prevSystemKernelTime == 0) // These may be 0 when we report CPU usage for the first time, in which case we should just return 0.
-            {
-                cpuUsage = 0;
-            }
-            else
+            if (Interop.Kernel32.GetProcessTimes(Interop.Kernel32.GetCurrentProcess(), out _, out _, out long procKernelTime, out long procUserTime) &&
+                Interop.Kernel32.GetSystemTimes(out _, out long systemUserTime, out long systemKernelTime))
             {
                 long totalProcTime = (procUserTime - s_prevProcUserTime) + (procKernelTime - s_prevProcKernelTime);
                 long totalSystemTime = (systemUserTime - s_prevSystemUserTime) + (systemKernelTime - s_prevSystemKernelTime);
-                cpuUsage = (int)(totalProcTime * 100 / totalSystemTime);
-            }
 
-            s_prevProcUserTime = procUserTime;
-            s_prevProcKernelTime = procKernelTime;
-            s_prevSystemUserTime = systemUserTime;
-            s_prevSystemKernelTime = systemKernelTime;
+                if (s_prevSystemUserTime != 0 && s_prevSystemKernelTime != 0 && // These may be 0 when we report CPU usage for the first time, in which case we should just return 0.
+                    totalSystemTime != 0)
```

Comment:
should make this condition `> 0` instead? just in case run into overflow?

Reply:
If GetSystemTimes or GetProcessTimes is giving us negative numbers (which I hope never happens) I think it would be beneficial for the cpuUsage value to be obviously bad rather than setting it to a benign looking value like zero.